crates/starknet/src/starknet/mod.rs

```diff
@@ -252,16 +252,44 @@ impl Starknet {
         Ok(())
     }
 
-    // TODO should return a more generic type (StateReader) to allow future implementation of a
-    // ForkedStateReader
-    pub fn get_state_reader_at(&self, block_id: &BlockId) -> Result<&InMemoryStateReader> {
+    fn get_state_at(&self, block_id: &BlockId) -> Result<&StarknetState> {
         match block_id {
-            BlockId::Tag(_) => Ok(&self.state.state),
+            BlockId::Tag(_) => Ok(&self.state),
             BlockId::Hash(_) => Err(Error::BlockIdHashUnimplementedError),
             BlockId::Number(_) => Err(Error::BlockIdNumberUnimplementedError),
         }
     }
 
+    pub fn get_class_hash_at(
+        &self,
+        block_id: &BlockId,
+        contract_address: &Address,
```

Comment:
there is already created type in types crate - ContractAddress. Ideally it should be used and inside the method converted to starknet_in_rust type Address

Reply:
Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/76/commits/15ed994521c504d98291a842eb9506c60c2e464f